apps/server/src/shared/domain/domainobject/board/card.do.ts

```diff
@@ -15,7 +16,7 @@ export class Card extends BoardComposite implements CardProps, BoardNodeBuildabl
 	}
 
 	addChild(child: AnyBoardDo, position?: number) {
-		if (child instanceof TextElement) {
+		if (child instanceof TextElement || child instanceof FileElement) {
```

Comment:
The childs should have the same base class / interface and this must be used for the input check. Or is this use as filter which elements are allowed to add on card level? Then please outsource the if check to a private method. "isAllowedChildInstance(child): boolean"

Reply:
The BoardComposite (superclass) defines an abstract method to add child, which has to be implemented by all composites. The purpose as you guessed is to give it opportunity to control what types of children are allowed to be added, and then delegate the actual add operation to its parent.
How to implement that is a detail for me, an isAllowedChildInstance does sound like a more readable way though... Ill take this into the team.